### PR TITLE
VolumetricFractionalMaxPooling like Spatial...

### PR DESCRIPTION
**Pull request by bottler**
We wanted a 3D version of Fractional Max Pooling and so I copied the code for SpatialFractionalMaxPooling and adapted it. I stayed as close as I could to the layout of the existing class. This code is all separate from the 2d version. I also have a similar change in cunn at https://github.com/bottler/cunn , but I haven't made a PR yet.

### CODE REVIEW

**Comment by soumith** [comment]
cc: @gchanan for review. Should be easy when diffed against SpatialFractionalMaxPooling

File: lib/THNN/generic/VolumetricFractionalMaxPooling.c [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+  long inputT = THTensor_(size)(input, timeDim);
+
+  THArgCheck(outputH + poolSizeH - 1 < inputH, 9,
+             "poolSizeH (%d) too large relative to input height (%d)",
+	     poolSizeH, inputH);
+  THArgCheck(outputW + poolSizeW - 1 < inputW, 8,
+             "poolSizeW (%d) too large relative to input width (%d)",
+	     poolSizeW, inputW);
+  THArgCheck(outputT + poolSizeT - 1 < inputT, 7,
+             "poolSizeW (%d) too large relative to input width (%d)",
```
Comment (gchanan) [inline lib/THNN/generic/VolumetricFractionalMaxPooling.c]: "poolSizeT (%d) too large relative to input time (%d)",

File: lib/THNN/generic/VolumetricFractionalMaxPooling.c [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+  for (plane = 0; plane < numPlanes; plane++) {
+    real* gradInputForPlane = gradInput + plane * inputT * inputW * inputH;
+    real* gradOutputForPlane = gradOutput + plane * outputT * outputW * outputH;
+    THIndex_t* indicesForPlane = indices + plane * outputT * outputW * outputH;
+
+    long h, w, t;
+    for (h = 0; h < outputH; ++h) {
+      for (w = 0; w < outputW; ++w) {
+        for (t = 0; t < outputT; ++t) {
+          long outputIndex = h * outputW * outputT + w * outputT+ t;
```
Comment (gchanan) [inline lib/THNN/generic/VolumetricFractionalMaxPooling.c]: nit: spacing off.

**Comment by bottler** [comment]
You're right on all those things. I've pushed another commit fixing them.

**Comment by nicholas-leonard** [comment]
@bottler Thanks. How is that cunn PR coming?

**Comment by bottler** [comment]
https://github.com/torch/cunn/pull/434

**Comment by bottler** [comment]
Thanks!

**Pull request merged**